source/common/ssl/context_impl.cc

```diff
@@ -209,7 +209,7 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
         ASN1_STRING* str = altname->d.uniformResourceIdentifier;
         char* crt_san = reinterpret_cast<char*>(ASN1_STRING_data(str));
```

Comment:
...it is. Feel free to ignore this.

Alternatively, you could use `ASN1_STRING_length()` and pass `size_t` to do the `crt_san_len >= uriPattern.length()` check and use `std::string::compare()`.